src/commonMain/kotlin/com/configcat/ConfigCatClient.kt

```diff
@@ -264,82 +272,84 @@ internal class Client private constructor(
             return details
         }
+        val setting = settingResult.settings[key]
 
-        val setting = result.settings[key]
-        if (setting == null) {
-            val message = "Value not found for key '$key'. Here are the available keys: " +
-                    result.settings.keys.joinToString(", ")
-            val details = EvaluationDetails.makeError(key, defaultValue, message, evalUser)
-            logger.error(message)
-            hooks.invokeOnFlagEvaluated(details)
-            return details
-        }
-
-        return evaluate(setting, key, evalUser, result.fetchTime)
+        return evaluate(setting!!, key, evalUser, settingResult.fetchTime)
     }
 
     override suspend fun getAllValueDetails(user: ConfigCatUser?): Collection<EvaluationDetails> {
-        val result = getSettings()
-        return result.settings.map {
-            evaluate(it.value, it.key, user ?: defaultUser, result.fetchTime)
+        val settingResult = getSettings()
+        if (!checkSettingsAvailable(settingResult, "empty list")) {
+            return emptyList()
+        }
+        return settingResult.settings.map {
+            evaluate(it.value, it.key, user ?: defaultUser, settingResult.fetchTime)
         }
     }
 
-    @Deprecated("This method is obsolete and will be removed in a future major version. " +
-            "Please use getValueDetails() instead.")
+    @Deprecated(
+        "This method is obsolete and will be removed in a future major version. " +
+            "Please use getValueDetails() instead."
+    )
     override suspend fun getVariationId(key: String, defaultVariationId: String?, user: ConfigCatUser?): String? {
         val result = getSettings()
         if (result.settings.isEmpty()) {
-            logger.error("Config JSON is not present. Returning defaultVariationId: '$defaultVariationId'.")
+            logger.error(
+                1000,
+                ConfigCatLogMessages.getConfigJsonIsNotPresentedWithDefaultValue(
+                    key,
+                    "defaultVariationId",
+                    defaultVariationId
+                )
+            )
             return defaultVariationId
         }
 
         val setting = result.settings[key]
         if (setting == null) {
-            logger.error("Value not found for key '$key'. Here are the available keys: " +
-                    result.settings.keys.joinToString(", "))
+            logger.error(
+                1001,
+                ConfigCatLogMessages.getSettingEvaluationFailedDueToMissingKey(
+                    key,
+                    "defaultVariationId",
+                    defaultVariationId,
+                    result.settings.keys
+                )
+            )
             return defaultVariationId
         }
 
         return evaluate(setting, key, user ?: defaultUser, result.fetchTime).variationId
     }
 
     override suspend fun getKeyAndValue(variationId: String): Pair<String, Any>? {
```

Comment:
Probably we should log error 2011 in case if the variation id is not found to be consistent with other SDKs. For example: https://github.com/configcat/java-sdk/blob/51b0de834759fff4f868c03337f63bcc90966a42/src/main/java/com/configcat/ConfigCatClient.java#L546